view/frontend/web/js/view/payment/omise-offsite-method-renderer.js

```diff
@@ -0,0 +1,69 @@
+define(
+    [
+        'Omise_Payment/js/view/payment/omise-offline-method-renderer',
```

Comment:
Just feel weird that offsite method renderer is extending its base from offline renderer.
Is there any reason why we shouldn't extend from `omise-base-method-renderer.js`? (besides sharing `isActive` function)

Reply:
If just only because it is sharing `isActive` behaviour then I'd like to propose to move 
`getOrderCurrency` and `getStoreCurrency` to `omise-base-method-renderer.js` then let's have `isActive` both on `omise-offline-method-renderer.js` and `omise-offsite-method-renderer.js `. 

Although the code will be duplicated but I think it more makes sense that `offsite` and `offline` should be extending its capability from `base-method-renderer`.

Reply:
Offsite extends from offline, Offline extends from `omise-base-method-renderer.js` - was the best way to keep code only where needed

Reply:
Programmatically, I would agree on this as there is no single code that has to be duplicated.

However, talking about an intention of the code as a reader perspective.
Reading this one and it could be implied that `Offsite` is either subset of `Offline` or have something related to `Offline` method while in fact, it just happens to have a similar behaviour on `isActive` function. Nothing more or less.

Reply:
Names could be better, could name 'omise-offline-method-renderer.js' as 'active-for-thb-method-renderer.js` or something like that. Then it's like we are combining traits to make up the final renderers

Reply:
Alternative idea, how about we move the whole functions in `offline-method-renderer.js` to `base`?

```js
/**
 * Is method available to display
 *
 * @return {boolean}
 */
isActive: function() {
    return this.getOrderCurrency() === 'thb' && this.getStoreCurrency() === 'thb';
},

/**
 * Get order currency
 *
 * @return {string}
 */
getOrderCurrency: function () {
    return window.checkoutConfig.quoteData.quote_currency_code.toLowerCase();
},

/**
 * Get store currency
 *
 * @return {string}
 */
getStoreCurrency: function () {
    return window.checkoutConfig.quoteData.store_currency_code.toLowerCase();
}
```

This whole part.
Then, `omise-cc-method.js` can have this 
```js
isActive: function() {
    return true;
}
```

Also we can move `placeOrder` from `omise-offline-tesco-method.js` to `offline-method-renderer.js`

Reply:
Seems like not a great idea as it would imply that all new payment methods by default are only for THB

Reply:
We could add a var something like `restrictedToCurrencies` which could contain an array of valid currencies for the payment method. Then we could have a single `isActive` based on this var (returns `true` if no restrictions)

Reply:
another one more alternative idea, how about this:
Move `getOrderCurrency`, `getStoreCurrency` from `offline-method-renderer.js` to `base-method-render.js` 
```js
// omise-base-method-renderer.js
define(
    [
        'Magento_Checkout/js/model/error-processor',
        'Magento_Checkout/js/model/full-screen-loader'
    ],
    function (
        errorProcessor,
        fullScreenLoader
    ) {
        'use strict';

        return {
            // ... the same code

            /**
             * Get order currency
             *
             * @return {string}
             */
            getOrderCurrency: function () {
                return window.checkoutConfig.quoteData.quote_currency_code.toLowerCase();
            },

            /**
             * Get store currency
             *
             * @return {string}
             */
            getStoreCurrency: function () {
                return window.checkoutConfig.quoteData.store_currency_code.toLowerCase();
            }
        };
    }
);
```

Then `offline-method-renderer.js` just
```js
define(
    [
        'Omise_Payment/js/view/payment/omise-base-method-renderer'
    ],
    function (
        Base
    ) {
        'use strict';

        return Object.assign({}, Base, {

            /**
             * Is method available to display
             *
             * @return {boolean}
             */
            isActive: function() {
                // ...
            },

            /**
             * Hook the placeOrder function.
             * Original source: placeOrder(data, event); @ module-checkout/view/frontend/web/js/view/payment/default.js
             *
             * @return {boolean}
             */
            placeOrder: function(data, event) {
                // ...
            }
        });
    }
);
```

The same as `offsite-method-renderer.js`
```js
define(
    [
        'Omise_Payment/js/view/payment/omise-base-method-renderer',
        'Magento_Checkout/js/model/url-builder',
        'mage/storage',
        'jquery'
    ],
    function (
        Base,
        urlBuilder,
        storage,
        $
    ) {
        'use strict';

        return Object.assign({}, Base, {

            /**
             * Is method available to display
             *
             * @return {boolean}
             */
            isActive: function() {
                // ...
            },

            /**
             * Hook the placeOrder function.
             * Original source: placeOrder(data, event); @ module-checkout/view/frontend/web/js/view/payment/default.js
             *
             * @return {boolean}
             */
            placeOrder: function (data, event) {
                // ...
            }
        });
    }
);
```

Reply:
Here we take `isActive` and `placeOrder` as sort of `interface` functions that all `payment-flow` should provide.
In this case, `offline`, `offsite`, `creditcard`

Reply:
> We could add a var something like restrictedToCurrencies which could contain an array of valid currencies for the payment method. Then we could have a single isActive based on this var (returns true if no restrictions)

Personally I think we don't really need to reduce down line of code to the point that it's super lean but don't feel right. But I think this would work too, if you prefer 👍